src/coder/encode.js

```diff
@@ -1,21 +1,20 @@
 'use strict'
 
-const { Buffer } = require('buffer')
 const varint = require('varint')
 const BufferList = require('bl/BufferList')
 
 const POOL_SIZE = 10 * 1024
 
 class Encoder {
   constructor () {
-    this._pool = Buffer.allocUnsafe(POOL_SIZE)
+    this._pool = new Uint8Array(POOL_SIZE)
```

Comment:
We should run the mega stress tests to verify the performance here is stable. https://github.com/libp2p/js-libp2p-interfaces/blob/master/src/stream-muxer/tests/mega-stress-test.js#L7